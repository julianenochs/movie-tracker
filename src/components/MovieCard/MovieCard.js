import React, { Component } from 'react';
import './MovieCard.scss';
import star from '../../images/star-white.svg';
import starFavorited from '../../images/star-favorited.svg';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { deleteFavorite, favorite, getFavorites } from '../../apiCalls';
import { updateFavorites } from '../../actions/index';

class MovieCard extends Component {

  refreshFavorites = async () => {
    const favorites = await getFavorites(this.props.user.userId);

    const update = await this.props.updateFavorites(favorites);
  }

  handleUnfavorite = async (e) => {
    const userId = this.props.user.userId;
    const movieId = Number(e.target.closest('section').id);
    const deleter = await deleteFavorite(userId, movieId);
    const test = await this.refreshFavorites();
  }

  handleFavorite = async (e) => {
    const userId = this.props.user.userId;
    const curMovieId = Number(e.target.closest('section').id);
    const movie = this.props.movies.find(movie => movie.id === curMovieId);
    const {id, title, poster_path, release_date, vote_average} = movie;
    const favorited = await favorite(userId, id, title, poster_path, release_date, vote_average, 'overview')
      .then(fav => fav);
    this.props.updateFavorites({favorites: [...this.props.favorites, favorited] });
  }

  selectMovie = e => {
    const id = e.target.closest('section').id;
    console.log(id);
  }

  render() {
    const { title, poster, overview, movieID, isLoggedIn, favorites } = this.props;
    const isFavorite = favorites.find(fav => fav.title === title);
    const starUrl = isFavorite ? starFavorited : star;
    const handleFavoriting = isFavorite ? this.handleUnfavorite : this.handleFavorite;
    return (
      <section className='movie__card' id={movieID}>
        <img
          src={`https://image.tmdb.org/t/p/original/${poster}`}
          alt={`movie poster for ${title}`}
          />
        <p className='overview'>{overview}</p>
        <NavLink to={`/movies/${movieID}`} className='view-movie__div'>
          <div >View Movie</div>
        </NavLink>
          <div>
            {isLoggedIn && <img className='favorite__star' src={starUrl} alt='favorite-button' onClick={handleFavoriting} />}
            {!isLoggedIn && (
              <NavLink to='/login'>
                <img src={star} className='favorite__star' alt='favorite-button' />
              </NavLink>
            )}
          </div>
      </section>
    );
  }
};

const mapStateToProps = state => ({
  isLoggedIn: state.isLoggedIn,
  favorites: state.favorites,
  user: state.user,
  movies: state.movies,
});

const mapDispatchToProps = dispatch => ({
  updateFavorites: favorites => dispatch( updateFavorites(favorites) ),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MovieCard);
