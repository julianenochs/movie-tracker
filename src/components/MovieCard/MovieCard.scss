$button-color: #004263;
$button-shadow: #003149;

img {
  width: 200px;
}

.movie__card {
  align-items: center;
  background-color: rgba(0, 31, 46, 0.78);
  color: white;
  display: flex;
  flex-direction: column;
  margin: 3px;
  margin-block-start: 5px;
  height: 70vh;
  padding: 15px;
  width: 15%;
}

.movie__title {
  font-family: 'Staatliches', cursive;
  font-size: 1.7em;
  height: 8vh;
  letter-spacing: 3px;
}

.overview {
  height: 20vh;
  font-family: 'Barlow Condensed', sans-serif;
  font-size: 1em;
  overflow: scroll;
  width: 95%;
}

.favorite__star {
  height: 4vh;
  width: 20%;
}

.view-movie__div {
  background-color: $button-color;
  border: none;
  box-shadow: 4px 4px $button-shadow;
  color: white;
  font-family: 'Barlow Condensed', sans-serif;
  font-size: 1em;
  height: 40px;
  letter-spacing: 5px;
  text-decoration: none;
  width: 120px;
}