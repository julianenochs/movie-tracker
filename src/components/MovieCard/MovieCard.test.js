import React from 'react';
import { shallow } from 'enzyme';
import MovieCard from './MovieCard';

describe('Movie Card', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<MovieCard

    />)
  })
  it.skip('should match the snapshot with all of the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  });

})