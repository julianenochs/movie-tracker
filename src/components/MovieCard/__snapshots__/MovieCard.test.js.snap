// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Card should match the snapshot with all of the data passed through 1`] = `undefined`;
